### ISSUE TITLE
Problems when interpreting useAs

### ISSUE THREAD

**Issue by LorenzoBettini** [LorenzoBettini/edelta#105]
When the interpreter instantiates an `EdeltaUseAs`, it uses the default constructor, thus the `EdeltaInterpreterEdeltaImpl` is not passed as the delegate library to search for EPackages (during interpretation). This is a problem if we use a reusable library function that refers to an EPackage with an ecoreref (currently we always refer to "ecore" which is always available).

Example, given this reusable function (in `edelta.example` project):

```
metamodel "ecore"
metamodel "myecore"

def createANewClassInMyEcore(String name) {
	ecoreref(myecore).addNewEClass(name)
}
```

if we call it from another file:

```
metamodel "ecore"
metamodel "myecore"

use ExampleReusableFunctions as myfunctions

modifyEcore SomeChanges epackage myecore {
	myfunctions.createANewClassInMyEcore("ANewClass")
}
```

we get a NPE in the interpreter, since when calling the reusable function from this file the `ecoreref(myecore)` is not resolved.

Of course the generated Java file is correct instead:

```java
public class Example extends AbstractEdelta {
  private ExampleReusableFunctions myfunctions;
  
  public Example() {
    myfunctions = new ExampleReusableFunctions(this);
  }
```

Since it passes `this` as the delegate.

**Comment by LorenzoBettini** [comment]
Moreover, `EdeltaInterpreterEdeltaImpl` just overrides `getEPackage` and if it's used as the `other` of another `AbstractEdelta` implementation this will not work as expected

**Comment by LorenzoBettini** [comment]
Closed by #106
